hsds/dset_lib.py

```diff
@@ -66,6 +67,41 @@ def getFillValue(dset_json):
     return arr
 
 
+def _get_arr_pts(arr_points, arr_index, pt, dim=0, chunk_index=None, factors=None):
+    """ recursive function to fill in chunk locations for hyperchunk selection.
+        arr_points: numpy array of shape (num_chunks, rank)
+        arr_index: row of arr_points for the first hyper chunk pt
+        pt: list of [0,] * rank - one element will get set for each recursive call
+        dim: the current dimension - only set in recursion
+        chunk_index: the HSDS chunk index
+        factors: the hyper scaling factors
+
+        on return arr_points rows arr_index[arr_index*N:(arr_index+1)*N-1] will be
+        set to the values needed to do a point selection on the chunk table, where
+        N is the ratioo of chunks to hyperchunks - np.prod(factors)
+
+    """
+
+    log.debug(f"get_arr_pts = arr_index: {arr_index}, dim: {dim}, pt: {pt}")
+    log.debug(f"chunk_index: {chunk_index}")
+
+    index = chunk_index[dim]
+    factor = factors[dim]
+    rank = len(chunk_index)
+    for i in range(factor):
+        pt[dim] = (index * factor + i)
+        if dim + 1 == rank:
+            idx = int(arr_index) + i
+            if rank == 1:
+                arr_points[idx] = pt[0]  # need to set 1d arrays with an int index
+            else:
+                arr_points[idx] = pt
+        else:
+            kwargs = {"dim": dim + 1, "chunk_index": chunk_index, "factors": factors}
+            next_index = arr_index + (i * np.prod(factors[1:]))
```

Comment:
Should this use `np.prod(factors[i:]` to account for all remaining dimensions? This might not matter right now since hyperchunking is limited to two dimensions at most, but it would be easier to understand and future-proof

Reply:
Yes this is a bug that shows up with more than two dimensions.  Fixed this and add a 3D test.  Should work for any number of dimensions now